src/main/java/core/basesyntax/service/strategy/strategyimpl/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (currentQuantity == null && currentQuantity < 0) {
+            throw new IllegalArgumentException("Invalid quantity of for purchase.");
+        }
+        int newQuantity = currentQuantity - purchaseQuantity;
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage - it should be positive. You need to add a check to ensure that `newQuantity` is not negative before proceeding to put it in the storage.